test/Dialect/FIRRTL/extract-instances.mlir

```diff
@@ -0,0 +1,342 @@
+        {circt.nonlocal = @nla_2, class = "circt.nonlocal"},
+        {circt.nonlocal = @nla_3, class = "circt.nonlocal"}
+      ]} @BBWrapper(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    firrtl.connect %out, %mod_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %mod_in, %in : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK-LABEL: firrtl.module @ExtractBlackBoxesSimple2
+  firrtl.module @ExtractBlackBoxesSimple2(in %in: !firrtl.uint<8>, out %out: !firrtl.uint<8>) {
+    // CHECK: %dut_in, %dut_out, %dut_prefix_0_in, %dut_prefix_0_out, %dut_prefix_1_in, %dut_prefix_1_out = firrtl.instance dut
+    // CHECK-NOT: annotations =
+    // CHECK-SAME: sym {{@.+}} @DUTModule
+    // CHECK-NEXT: %bb_in, %bb_out = firrtl.instance bb sym [[BB_SYM:@.+]] {annotations = [{class = "DummyB"}, {class = "DummyC"}, {circt.nonlocal = @nla_1, class = "circt.nonlocal"}]} @MyBlackBox
+    // CHECK-NEXT: firrtl.strictconnect %bb_in, %dut_prefix_1_in
+    // CHECK-NEXT: firrtl.strictconnect %dut_prefix_1_out, %bb_out
+    // CHECK-NEXT: %bb2_in, %bb2_out = firrtl.instance bb2 sym [[BB2_SYM:@.+]] @MyBlackBox2
+    // CHECK-NEXT: firrtl.strictconnect %bb2_in, %dut_prefix_0_in
+    // CHECK-NEXT: firrtl.strictconnect %dut_prefix_0_out, %bb2_out
+    %dut_in, %dut_out = firrtl.instance dut sym @dut {annotations = [
+        {circt.nonlocal = @nla_3, class = "circt.nonlocal"}
+      ]} @DUTModule(in in: !firrtl.uint<8>, out out: !firrtl.uint<8>)
+    firrtl.connect %out, %dut_out : !firrtl.uint<8>, !firrtl.uint<8>
+    firrtl.connect %dut_in, %in : !firrtl.uint<8>, !firrtl.uint<8>
+  }
+  // CHECK: sv.verbatim "
+  // CHECK-SAME{LITERAL}: prefix_0 -> {{0}}.{{1}}.{{2}}\0A
+  // CHECK-SAME{LITERAL}: prefix_1 -> {{0}}.{{1}}.{{3}}\0A
+  // CHECK-SAME: output_file = #hw.output_file<"BlackBoxes.txt", excludeFromFileList>
+  // CHECK-SAME: symbols = [
+  // CHECK-SAME: @DUTModule
+  // CHECK-SAME: #hw.innerNameRef<@DUTModule::[[WRAPPER_SYM]]>
+  // CHECK-SAME: #hw.innerNameRef<@ExtractBlackBoxesSimple2::[[BB2_SYM]]>
+  // CHECK-SAME: #hw.innerNameRef<@ExtractBlackBoxesSimple2::[[BB_SYM]]>
+  // CHECK-SAME: ]
+}
+
+//===----------------------------------------------------------------------===//
+// ExtractBlackBoxes IntoDUTSubmodule
+//===----------------------------------------------------------------------===//
+
+// CHECK: firrtl.circuit "ExtractBlackBoxesIntoDUTSubmodule"
+firrtl.circuit "ExtractBlackBoxesIntoDUTSubmodule"  {
+  // CHECK-LABEL: firrtl.nla @nla_1 [
+  // CHECK-SAME:    #hw.innerNameRef<@ExtractBlackBoxesIntoDUTSubmodule::@tb>
+  // CHECK-SAME:    #hw.innerNameRef<@TestHarness::@dut>
+  // CHECK-SAME:    #hw.innerNameRef<@DUTModule::@BlackBoxes>
+  // CHECK-SAME:    #hw.innerNameRef<@BlackBoxes::@bb1>
+  // CHECK-SAME:  ]
+  firrtl.nla @nla_1 [#hw.innerNameRef<@ExtractBlackBoxesIntoDUTSubmodule::@tb>, #hw.innerNameRef<@TestHarness::@dut>, #hw.innerNameRef<@DUTModule::@mod>, #hw.innerNameRef<@BBWrapper::@bb1>]
```

Comment:
For these massive tests, consider also splitting the NLA paths across multiple lines like with the checks.